modtools/intro-loader.lua

```diff
@@ -0,0 +1,160 @@
+--Script for making randomized, entity-specific intros on embark
+--Written by Khaos with input from Amostubal
+
+local help = [====[
+intro-loader.lua
+===============
+
+Alters the fort mode intro text before embarking
+
+This script should be called in "dfhack.init" for proper operation. Attempting
+ to call the script at any other point in execution will have unpredictable
+ results.
+
+This script should not be called from the terminal outside of debug usage.
+
+The script requires either the -entity, -rand, or both parameters be included
+ in the call to operate, and will indicate failure without
+
+
+Usage:
+
+-help
+    print this text
+-v
+    Activate verbose mode
+-folder
+    Designate a folder to get the input intro files from relative to
+    the DF directory
+    Defaults to /data/announcement/ if not specified
+-now
+    Run the script now, as opposed to waiting for the correct timing
+    For debug use, don't implement in init files.
+-file
+    Change the prefix for input filenames.
+    By default, the prefix is "INTRO", as in "INTRO_MOUNTAIN_1"
+-entity
+    Enables entity-specific intro files
+-rand
+    Enables random selection for intro files
+
+
+Introduction files have to be set up by the user
+Intro files must
+    -Be in the custom format used by Toady, editable via the WTF tools
+     developed by Andux. Available as of now at:
+        http://dffd.bay12games.com/file.php?id=4175
+    -In plaintext, the file must begin with the word "fortressintro" followed
+     by a newline. The first line must be exactly such, or it will fail to be
+     loaded by the game.
+    -The file must be named according to your chosen mode for the script:
+      - Parts of the filename are separated by underscores for readability
+        purposes
+      - Filenames must start with the defined prefix, which defaults to INTRO,
+        but can be changed with the -file parameter
+      - If the -entity argument is supplied, the entity name must be appeneded
+        to the prefix, as in INTRO_MOUNTAIN for dwarves
+      - If the -rand argument is supplied, the filenames must each have a unique
+        number appended, as in INTRO_MOUNTAIN_2
+      - Further, for the -rand argument, usable numbers start at 1 and all
+	    numbers must be consecutive.
+      - The files must not have file extensions
+]====]
+
+
+local utils = require 'utils'
+
+
+validArgs = validArgs or utils.invert({
+    'help',
+    'v'
+    'now',
+    'folder',
+    'file',
+    'rand',
+    'entity',
+})
+
+local args = utils.processArgs({...}, validArgs)
+
+if args.help then
+    print(help)
+    return
+end
+
+local inputfolder = dfhack.getDFPath() .. "\\data\\announcement\\"
+if args.folder then
+    inputfolder = dfhack.getDFPath() ..  args.folder .. "\\"
+end
+
+local prefix = "INTRO"
+if args.file then
+    prefix = args.file
+end
+
+if args.rand then
+    math.randomseed(os.time())
```

Comment:
I mean, is that not what the DFhack API is really doing, except maybe calling srand() in C?
I just find it weird to *not* make a call to a seeding function when I intend to deal with random numbers, and it's just always convenient to use OS time for it since that's always changing.
Especially with a function like this where lua's normal behavior of starting every function with a fixed seed would result in you never getting a different result for your intros.